### PR TITLE
Spike db connection pool retry

### PR DESCRIPTION
**Pull request by NickGrantX**
# Checklist
Reminder: If any change made to this repo affects the acceptance tests then the version pin in the Pipfile in [census-rm-acceptance-tests]() needs to be updated.
* [ ] Updated census-rm-acceptance-tests version pin (if applicable)

# Motivation and Context
The bulk processors can fail due to the connection being forced to time out after 30 seconds, which protects against the read replica drifting too far out of data versus the target database it mirrors.

# What has changed
Use a connection pool instead of a single DB connection, which has the capability of making new DB connections as required. Added retries, so that any glitches shouldn't cause the whole batch to fail.

# How to test?
Create a big batch which will use the DB to get data and/or validate (e.g. bulk QID invalidate) then run the batch, ensuring it takes at least a minute or more.

# Links
Trello: https://trello.com/c/DlW1kOIb

### CODE REVIEW

**Comment by NickGrantX** [comment]
[Fix bulk processors from timing out after 30 seconds](https://trello.com/c/DlW1kOIb/1659-fix-bulk-processors-from-timing-out-after-30-seconds)

**Review by liamtoozer — APPROVED** [review]
LGTM 👍 Tested in my GCP

**Review by hbrace — APPROVED** [review]
Also tested in my GCP and it works 👍

File: toolbox/utilities/db_helper.py [hunk 1]
```diff
@@ -17,27 +21,43 @@ def execute_sql_query(sql_query, db_host=Config.DB_HOST, extra_options=""):
...
-        conn.close()
+        if conn_pool:
+            conn_pool.closeall()
 
 
-def execute_in_connection(*args, conn=None):
-    cursor = conn.cursor()
-    cursor.execute(*args)
-    return cursor.fetchall()
+@retry(wait_exponential_multiplier=1000, wait_exponential_max=10000)
```
Comment (AdamHawtin) [inline toolbox/utilities/db_helper.py]: I believe with these settings it'll retry forever, should we have a max attempts?

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
Are we ok with it infinitely looking on any exceptions in the "execute in database" functions? I think we'd want a max retries or timeout.

File: toolbox/utilities/db_helper.py (hunk 1 above)
Comment (NickGrantX) [reply toolbox/utilities/db_helper.py]: yes, great idea

**Review by AdamHawtin — APPROVED** [review]
👍 Beautiful

**Pull request merged**